x-pack/platform/plugins/shared/fleet/server/services/epm/agent/agent.ts

```diff
@@ -15,14 +15,25 @@ import { toCompiledSecretRef } from '../../secrets';
 import { PackageInvalidArchiveError } from '../../../errors';
 import { appContextService } from '../..';
 
+import {
+  getHandlebarsCompiledTemplateCache,
+  setHandlebarsCompiledTemplateCache,
+} from '../packages/cache';
+
 const handlebars = Handlebars.create();
 
 export function compileTemplate(variables: PackagePolicyConfigRecord, templateStr: string) {
   const logger = appContextService.getLogger();
   const { vars, yamlValues } = buildTemplateVariables(logger, variables);
   let compiledTemplate: string;
   try {
-    const template = handlebars.compile(templateStr, { noEscape: true });
+    let template = getHandlebarsCompiledTemplateCache(templateStr);
+
+    if (!template) {
+      template = handlebars.compile(templateStr, { noEscape: true });
```

Comment:
this seems a big change, as for 100 agent policies, this seems to be called at least 200 times per template.